File: policy-pulse-desktop/setup_env.js
/**
 * Policy Pulse Desktop - Environment Setup Script
 *
 * This script automates the setup of the Node.js environment for Policy Pulse Desktop.
 */

const { execSync } = require('child_process');
const fs = require('fs');
const path = require('path');

function setupNodeEnvironment() {
  console.log('Setting up Node.js environment for Policy Pulse Desktop...');

  try {
    // Check if package.json exists
    if (!fs.existsSync('./package.json')) {
      console.error('✗ package.json not found');
      return false;
    }

    // Install dependencies
    console.log('Installing dependencies...');
    execSync('npm install', { stdio: 'inherit' });

    console.log('✓ Dependencies installed successfully');

    // Verify installation
    console.log('Verifying installation...');
    const packageJson = JSON.parse(fs.readFileSync('./package.json', 'utf8'));
    console.log(`✓ Package: ${packageJson.name}@${packageJson.version}`);

    return true;
  } catch (error) {
    console.error('✗ Failed to setup Node.js environment:', error.message);
    return false;
  }
}

function verifySetup() {
  console.log('Verifying Node.js setup...');

  try {
    // Check if required files exist
    const requiredFiles = ['package.json', 'node_modules'];
    for (const file of requiredFiles) {
      if (!fs.existsSync(file)) {
        console.error(`✗ Required file/directory missing: ${file}`);
        return false;
      }
    }

    console.log('✓ All required files present');

    // Try to run a basic check
    execSync('npm list', { stdio: 'ignore' });
    console.log('✓ npm dependencies verified');

    return true;
  } catch (error) {
    console.error('✗ Setup verification failed:', error.message);
    return false;
  }
}

function main() {
  console.log('Policy Pulse Desktop Environment Setup');
  console.log('=' .repeat(40));

  let success = true;

  // Setup Node.js environment
  if (!setupNodeEnvironment()) {
    success = false;
  }

  // Verify setup
  if (success && !verifySetup()) {
    success = false;
  }

  if (success) {
    console.log('\n✓ Setup completed successfully!');
    console.log('\nTo run the application:');
    console.log('  npm start');
    console.log('\nTo build for production:');
    console.log('  npm run build');
    console.log('\nTo run tests:');
    console.log('  npm test');
  } else {
    console.log('\n✗ Setup failed. Please check the error messages above.');
    process.exit(1);
  }

  return 0;
}

if (require.main === module) {
  process.exit(main());
}

module.exports = { setupNodeEnvironment, verifySetup };

File: policy-pulse-core/setup_env.py
#!/usr/bin/env python3
"""
Policy Pulse Core - Environment Setup Script

This script automates the setup of the Python environment for Policy Pulse Core.
"""

import os
import sys
import subprocess
import argparse
from pathlib import Path


def setup_virtual_environment():
    """Set up Python virtual environment."""
    print("Setting up Python virtual environment...")

    try:
        # Create virtual environment
        subprocess.run([sys.executable, "-m", "venv", "venv"], check=True)
        print("✓ Virtual environment created successfully")
        return True
    except subprocess.CalledProcessError as e:
        print(f"✗ Failed to create virtual environment: {e}")
        return False


def install_dependencies():
    """Install project dependencies."""
    print("Installing dependencies...")

    try:
        # Activate virtual environment and install dependencies
        if os.name == 'nt':  # Windows
            pip_path = os.path.join("venv", "Scripts", "pip")
        else:  # Unix-like systems
            pip_path = os.path.join("venv", "bin", "pip")

        subprocess.run([pip_path, "install", "-r", "requirements.txt"], check=True)
        print("✓ Dependencies installed successfully")
        return True
    except subprocess.CalledProcessError as e:
        print(f"✗ Failed to install dependencies: {e}")
        return False


def verify_setup():
    """Verify that the setup was successful."""
    print("Verifying setup...")

    try:
        # Try importing core modules
        import policy_pulse_core
        print("✓ Core modules imported successfully")
        return True
    except ImportError as e:
        print(f"✗ Failed to import core modules: {e}")
        return False


def main():
    """Main setup function."""
    parser = argparse.ArgumentParser(description="Setup Policy Pulse Core environment")
    parser.add_argument("--skip-venv", action="store_true", help="Skip virtual environment creation")
    parser.add_argument("--skip-deps", action="store_true", help="Skip dependency installation")

    args = parser.parse_args()

    print("Policy Pulse Core Environment Setup")
    print("=" * 40)

    # Setup steps
    success = True

    if not args.skip_venv:
        if not setup_virtual_environment():
            success = False

    if not args.skip_deps and success:
        if not install_dependencies():
            success = False

    if success:
        print("\n✓ Setup completed successfully!")
        print("\nTo activate the environment:")
        if os.name == 'nt':  # Windows
            print("  venv\\Scripts\\activate")
        else:  # Unix-like systems
            print("  source venv/bin/activate")
        print("\nTo verify installation:")
        print("  python -c \"import policy_pulse_core; print('Success!')\"")
    else:
        print("\n✗ Setup failed. Please check the error messages above.")
        return 1

    return 0


if __name__ == "__main__":
    sys.exit(main())